qubesmanager/qube_manager.py

```diff
@@ -135,21 +135,31 @@ def __init__(self, vm, row_no, table):
 
         self.table = table
 
-    def update(self, update_size_on_disk=False):
+    def update(self, update_size_on_disk=False, event=None):
         """
         Update info in a single VM row
         :param update_size_on_disk: should disk utilization be updated? the
         widget will extract the data from VM object
+        :param event: name of the event that caused the update, to avoid
+        updating unnecessary properties; if event is none, update everything
         :return: None
         """
         try:
             self.info_widget.update_vm_state()
-            self.template_widget.update()
-            self.netvm_widget.update()
-            self.internal_widget.update()
-            self.ip_widget.update()
-            self.include_in_backups_widget.update()
-            self.last_backup_widget.update()
+            if not event or event.endswith('label'):
+                self.label_widget.update()
+            if not event or event.endswith('template'):
+                self.template_widget.update()
+            if not event or event.endswith('netvm'):
+                self.netvm_widget.update()
+            if not event or event.endswith('internal'):
+                self.internal_widget.update()
+            if not event or event.endswith('ip'):
+                self.ip_widget.update()
+            if not event or event.endswith('include_in_backups'):
+                self.include_in_backups_widget.update()
+            if not event or event.endswith('backup_timestamp'):
+                self.last_backup_widget.update()
```

Comment:
Theoretically this will lead to extra updates if name of one property is a suffix of another. You should match even name including ":" separator.

Reply:
fixed; I don't know why github is not showing this as changed (maybe because I amended a commit).